src/orange/detail/BIHBuilder.cc

```diff
@@ -18,59 +18,68 @@ namespace detail
 {
 //---------------------------------------------------------------------------//
 /*!
- * Construct from a Storage object
+ * Construct from a Storage object.
  */
-BIHBuilder::BIHBuilder(Storage* storage) : storage_(storage)
+BIHBuilder::BIHBuilder(Storage* storage)
+    : bboxes_{&storage->bboxes}
+    , local_volume_ids_{&storage->local_volume_ids}
+    , inner_nodes_{&storage->inner_nodes}
+    , leaf_nodes_{&storage->leaf_nodes}
 {
-    CELER_EXPECT(storage_);
+    CELER_EXPECT(storage);
 }
 
 //---------------------------------------------------------------------------//
 /*!
  * Create BIH Nodes.
  */
-BIHTree BIHBuilder::operator()(VecBBox bboxes)
+BIHTree BIHBuilder::operator()(VecBBox&& bboxes)
 {
     CELER_EXPECT(!bboxes.empty());
 
     // Store bounding boxes and their corresponding centers
-    bboxes_ = std::move(bboxes);
-    centers_.resize(bboxes_.size());
-    std::transform(bboxes_.begin(),
-                   bboxes_.end(),
-                   centers_.begin(),
+    temp_.bboxes = std::move(bboxes);
+    temp_.centers.resize(temp_.bboxes.size());
+    std::transform(temp_.bboxes.begin(),
+                   temp_.bboxes.end(),
+                   temp_.centers.begin(),
                    &celeritas::calc_center<fast_real_type>);
 
     // Separate infinite bounding boxes from finite
     VecIndices indices;
     VecIndices inf_volids;
-    for (auto i : range(bboxes_.size()))
+    for (auto i : range(temp_.bboxes.size()))
     {
         LocalVolumeId id(i);
 
-        if (!bboxes_[i])
+        if (!temp_.bboxes[i])
         {
             // Null bbox (background volume) is unreachable by volume
             // initialization
         }
-        else if (is_infinite(bboxes_[i]))
+        else if (is_infinite(temp_.bboxes[i]))
         {
+            // Infinite in *every* direction
+            // TODO: make an exception for "EXTERIOR" volume and remove the
+            // "infinite volume" exceptions?
```

Comment:
After this refactor, *only* the exterior (well, and the stuff coming from SCALE for the moment) should have an infinite volume.